core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -242,11 +248,16 @@ public IndexShard(ShardId shardId, @IndexSettings Settings indexSettings, ShardP
         this.flushThresholdSize = indexSettings.getAsBytesSize(INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE, new ByteSizeValue(512, ByteSizeUnit.MB));
         this.disableFlush = indexSettings.getAsBoolean(INDEX_TRANSLOG_DISABLE_FLUSH, false);
         this.indexShardOperationCounter = new IndexShardOperationCounter(logger, shardId);
+        this.indexingMemoryController = provider.getIndexingMemoryController();
+
         this.searcherWrapper = provider.getIndexSearcherWrapper();
         this.percolatorQueriesRegistry = new PercolatorQueriesRegistry(shardId, indexSettings, queryParserService, indexingService, mapperService, indexFieldDataService);
         if (mapperService.hasMapping(PercolatorService.TYPE_NAME)) {
             percolatorQueriesRegistry.enableRealTimePercolator();
         }
+
+        // TODO: can we somehow call IMC.forceCheck here?  Since we just became active, it can divvy up the RAM
```

Comment:
> I assume you tried? what broke?

I didn't try!  It just looked not simple because here we are in this `IndexShard` ctor and if we invoke IMC at this point, probably the caller/Guice (creating this shard, IndicesService maybe?) won't even have this shard listed in its state?

Maybe up above somewhere I could call it ... I'll try.